src/client/tile.cpp

```diff
@@ -2048,3 +2051,36 @@ video::SColor TextureSource::getTextureAverageColor(const std::string &name)
 	c.setAlpha(255);
 	return c;
 }
+
+video::ITexture *TextureSource::getShaderFlagsTexture(TileDef *tiledef, TileSpec *tile)
+{
+	video::ITexture *flags_texture;
+	std::string tname = "__shaderFlagsTexture";
+	std::string strue = "1";
+	std::string sfalse = "0";
+
+	bool normalmap_present = tile->normal_texture ? true : false;
+	tname += normalmap_present ? strue : sfalse;
+	tname += tiledef->tileable_horizontal ? strue : sfalse;
+	tname += tiledef->tileable_vertical ? strue : sfalse;
+
+	if (isKnownSourceImage(tname)) {
+		u32 id;
+		flags_texture = getTexture(tname.c_str(), &id);
+		return flags_texture;
+	} else {
+		video::IVideoDriver *driver = m_device->getVideoDriver();
+		video::IImage *flags_image = driver->createImage(
+			video::ECF_A8R8G8B8, core::dimension2d<u32>(1, 1));
+		video::SColor c(
+			255,
+			normalmap_present ? 255 : 0,
+			tiledef->tileable_horizontal ? 255 : 0,
+			tiledef->tileable_vertical ? 255 : 0);
+		flags_image->setPixel(0, 0, c);
```

Comment:
You never checked the return value of createImage().  If it fails and returns NULL, the entire program will crash.